scipy/stats/_continuous_distns.py

```diff
@@ -4076,6 +4076,101 @@ def fit(self, data, *args, **kwds):
 laplace = laplace_gen(name='laplace')
 
 
+class laplace_asymmetric_gen(rv_continuous):
+    r"""An asymmetric Laplace continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `laplace_asymmetric` is
+
+    .. math::
+
+        f(x, \kappa) &= \frac{1}{\kappa+\kappa^{-1}}\exp(-x\kappa),\quad x\ge0\\
+                     &= \frac{1}{\kappa+\kappa^{-1}}\exp(x/\kappa),\quad x<0\\
+
+    for :math:`-\infty < x < \infty`, :math:`\kappa > 0`.
+
+    `laplace_asymmetric` takes ``kappa`` as a shape parameter for
+    :math:`\kappa`. For :math:`\kappa = 1`, it is identical to a
+    Laplace distribution.
+
+    %(after_notes)s
+
+    See Also
+    --------
+    laplace : Laplace distribution
+
+    References
+    ----------
+    .. [1] Kozubowski TJ and Podg\'orski K. A Multivariate and
+           Asymmetric Generalization of Laplace Distribution,
+           Computational Statistics 15, 531--540 (2000).
+           :doi:`10.1007/PL00022717`
+
+    %(example)s
+
+    """
+    def _rvs(self, kappa, size=None, random_state=None):
+        kapinv = 1/kappa
+        u = -kappa + (kappa+kapinv)*random_state.uniform(size=size)
+        return -np.where(u >= 0,
+            kapinv*np.log1p(-u*kappa),
+            -kappa*np.log1p(u*kapinv))
+
+    def _pdf(self, x, kappa):
+        return np.exp(self._logpdf(x, kappa))
+
+    def _logpdf(self, x, kappa):
+        kapinv = 1/kappa
+        lPx = np.where(x >= 0, -x*kappa, x*kapinv)
+        lPx -= np.log(kappa+kapinv)
+        return lPx
+
+    def _cdf(self, x, kappa):
+        kapinv = 1/kappa
+        kappkapinv = kappa+kapinv
+        return np.where(x >= 0,
+            1-np.exp(-x*kappa)*(kapinv/kappkapinv),
+            np.exp(x*kapinv)*(kappa/kappkapinv))
+
+    def _sf(self, x, kappa):
+        kapinv = 1/kappa
+        kappkapinv = kappa+kapinv
+        return np.where(x >= 0,
+            np.exp(-x*kappa)*(kapinv/kappkapinv),
+            1-np.exp(x*kapinv)*(kappa/kappkapinv))
+
+    def _ppf(self, q, kappa):
+        kapinv = 1/kappa
+        kappkapinv = kappa+kapinv
+        return np.where(q >= kappa/kappkapinv,
+            -np.log((1-q)*kappkapinv*kappa)*kapinv,
+            np.log(q*kappkapinv/kappa)*kappa)
```

Comment:
PEP-8 [indentation](https://www.python.org/dev/peps/pep-0008/#indentation) and whitespace:
```suggestion
        return np.where(q >= kappa/kappkapinv,
                        -np.log((1 - q)*kappkapinv*kappa)*kapinv,
                        np.log(q*kappkapinv/kappa)*kappa)
```

Reply:
Done thanks.